src/components/Identity/customerIds/createCustomerIds.js

```diff
@@ -34,7 +34,9 @@ export default (cookieJar, lifecycle, network, optIn) => {
   };
 
   const makeServerCall = payload => {
-    return network.sendRequest(payload, payload.expectsResponse);
+    return lifecycle.onBeforeDataCollection(payload).then(() => {
```

Comment:
This whole flow might change because setCustomerIDs might not be considered an event to start with. So the createCustomerIds might get injected with the ECID getter, and it adds it instead of calling the lifecycle.